src/workerd/io/worker-fs.c++

```diff
@@ -0,0 +1,245 @@
+#include "worker-fs.h"
+
+#include <workerd/io/io-context.h>
+
+namespace workerd {
+
+#define V(name, path) const jsg::Url FsMap::kDefault##name##Path = path##_url;
+KNOWN_VFS_ROOTS(V)
+#undef V
+
+namespace {
+
+// Thread-local storage to track the current temp directory storage scope
+// on the stack.
+static thread_local TempDirStoreScope* tempDirStorageScope = nullptr;
+
+class TempDirectory final: public jsg::fs::Directory {
+ public:
+  void setLastModified(jsg::Lock& js, kj::Date date = kj::UNIX_EPOCH) override {
```

Comment:
I don't think this should have a default value, if it should than, we should have a separate function called setLastModifiedToNow()

```suggestion
  void setLastModified(jsg::Lock& js, kj::Date date) override {
```